rapid7/metasploit-framework
# Sonicwall rce CVE-2023-34124

Add a module for a remote code execution issue in SonicWall GMS. This works by exploiting several different issues fixed in the same patch, culminating in shell injection. One of the biggest difficulties was getting past their XSS filter, which takes exception to a lot of special characters. Both the Windows and Linux versions should be able to encode/run any command that's not too long.

## Verification

The affected software is SonicWall GMS 9.3.9320 (and likely earlier). The first patched version is 9.3.9330. They should be available in the "vulnerable software" folder, but need to be linked to a (free) SonicWall account + demo license. I documented how to get those going in the `documentation/` entry.

This should work against both the Windows version (which is a .exe installer), and a Linux version (which is a VM appliance, which, AFAICT, they don't give you a login for, or at least not one I could find).

Once the software is installed:

1. Start `msfconsole`
2. Do `use exploit/multi/http/sonicwall_shell_injection_cve_2023_34124`
3. Set the appropriate `TARGET`, `RHOST`, `LHOST`, and possibly `FETCH_SRVHOST` (for Windows)
4. Do `run`
5. You should get meterpreter

Happy to help with running the software or testing the modules!
